crates/rpc-server/src/registry.rs

```diff
@@ -109,9 +106,9 @@ impl Registry {
             .with_method("compute_l2_sudt_script_hash", compute_l2_sudt_script_hash);
 
         // Tests
-        if TestMode::Enable == self.test_mode {
+        if let Some(r) = self.tests_rpc_impl {
             server = server
-                .with_data(Data(Arc::clone(&self.tests_rpc_impl)))
+                .with_data(Data(Arc::clone(&r)))
```

Comment:
```suggestion
                .with_data(Data(Arc::clone(&tests_rpc_impl)))
```

`r` is a meaningless name.